code/modules/reagents/Chemistry-Goon-Medicine.dm

```diff
@@ -0,0 +1,335 @@
+#define SOLID 1
+#define LIQUID 2
+#define GAS 3
+
+#define REM REAGENTS_EFFECT_MULTIPLIER
+
+datum/reagent/silver_sulfadiazine
+	name = "Silver Sulfadiazine"
+	id = "silver_sulfadiazine"
+	description = "100% chance per cycle of healing 2 points of BURN damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	metabolization_rate = 2
+
+datum/reagent/silver_sulfadiazine/reaction_mob(var/mob/living/M as mob, var/method=TOUCH, var/volume)
+	if(method == TOUCH)
+		M.adjustFireLoss((volume-(volume*2))*REM)
+		M << "<span class='notice'>You feel your burns healing!</span>"
+		M.emote("scream")
+	if(method == INGEST)
+		M.adjustToxLoss(0.5*volume)
+		M << "<span class='notice'>You probably shouldn't have eaten that. Maybe you should of splashed it on, or applied a patch?</span>"
+	..()
+	return
+
```

Comment:
Silver sulfa and Styptic both heal 1 damage per volume and not 2.

The tox damage is a once off thing, which you seem to have gotten down.

The per cycle effects of styptic and silver sulfa both have a 55% chance of healing 2 of their respective damage type; this is particularly useful for hyposprays on Goon as they avoid INGEST and it allows these chems to be used as unreliable injectables.